docs/zkapps/how-to-write-a-zkapp-ui.mdx

```diff
@@ -145,12 +137,99 @@ To see more details on deploying to Vercel, [see their documentation on `vercel.
 }
 ```
 
-**Setting COOP and COEP in your Cloudflare Pages configuration**:
+**Cloudflare Pages**
 
-To see more details on deploying to Cloudflare Pages, [see their documentation on `_headers`](https://developers.cloudflare.com/pages/platform/headers/).
+If your app will be hosted on Cloudflare Pages, you can set these headers via a [`_headers` file](https://developers.cloudflare.com/pages/platform/headers/).
 
 ```
 /*
   Cross-Origin-Opener-Policy: same-origin
   Cross-Origin-Embedder-Policy: require-corp
 ```
+
+### Connecting your zkApp with a user's wallet
+
+:::info
+
+Currently, only Auro Wallet's Chrome extension supports zkApp transactions. We
+anticipate other wallets to add support over time, which they can do by using
+the <a
+href="https://www.npmjs.com/package/@aurowallet/mina-provider">`auro-web3-provider`</a>
+from NPM within their browser-extension wallet.
+
+:::
+
+1. Install <a
+   href="https://chrome.google.com/webstore/detail/auro-walletmina-protocol/cnmamaachppnkjgnildpdmkaakejnhae">Auro
+   Wallet for Chrome</a>. Users of your zkApp will need to have this wallet
+   installed in order to interact with your zkApp. Once installed, `window.mina`
+   will automatically become available in the user's browser environment. Your
+   zkApp will then use this object to interact with the wallet.
+
+2. Get accounts - To fetch a user's list of Mina accounts, you can use the
+   `requestAccounts()` method. This can be useful to indicate if the user's wallet is successfully connected to your zkApp, such as in the screenshot below.
+
+```ts
+let accounts;
+
+try {
+  // Accounts is an array of string Mina addresses.
+  accounts = await window.mina.requestAccounts();
+} catch (err) {
+  // If the user has a wallet installed but has not created an account, an
+  // exception will be thrown. Consider showing "not connected" in your UI.
+  console.log(err.message);
+}
+
+// Show first 6 and last 4 characters of user's Mina account.
```

Comment:
Should be in the try block, otherwise the array access could fail if the provider returns a unexpected value